File: submissions/0792-binary-search/solution.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        int mid;
        int currentVal;
        
        while (left <= right) {
            mid = left + (right-left) / 2;
            currentVal = nums[mid];
            if(currentVal==target) {
                return mid;
            }
            if(target < currentVal) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return -1;
    }
}

File: submissions/0792-binary-search/solution.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        lo = -1
        hi = len(nums)
        while hi - lo > 1:
            mid = (lo+hi)//2
            val = nums[mid]
            if target < val:
                hi = mid
            else:
                lo = mid

        # nums[lo] <= target
        # nums[hi] > target
        return -1 if lo == -1 or nums[lo] != target else lo

